cp3pt0-deployment/common/migrate_singleton.sh

```diff
@@ -87,26 +88,77 @@ function main() {
     success "Migration is completed for Cloud Pak 3.0 Foundational singleton services."
 }
 
+function isolate_license_service_reporter(){
+    title "Isolating License Service Reporter"
+
+    local lsr_cr=$("${OC}" get IBMLicenseServiceReporter -A --no-headers)
+    local count=$(echo "$lsr_cr" | wc -l)
+
+    if [[ count -eq 0 ]]; then
+        info "No LSR for migration found in cluster"
+        return 0
+    fi
+
+    if [[ count -ne 1 ]]; then
+        info "Expecting exactly one IBMLicenseServiceReporter in cluster.${count} found."
+        return 0
+    fi
+
+    local ns=$(echo "$lsr_cr" | cut -d ' ' -f1)
+
+    local return_value=$( ("${OC}" get crd ibmlicenseservicereporters.operator.ibm.com > /dev/null && echo exists) || echo fail)
+    if [[ $return_value == "exists" ]]; then
+
+        return_value=$("${OC}" get ibmlicenseservicereporters -A --no-headers | wc -l)
```

Comment:
Are we able to move the LSR number counting logic down to here, after we check LSR CRD?

```
return_value=$("${OC}" get ibmlicenseservicereporters -A --no-headers | wc -l)
if [[ $return_value -eq 0 ]]; then
   return 0
fi
if [[ count -ne 1 ]]; then
   info "Expecting exactly one IBMLicenseServiceReporter in cluster.${count} found."
   return 0
fi

# continue change PVC to Retain
```